lib/Driver/ToolChains.cpp

```diff
@@ -1115,8 +1115,13 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   // Add the linker script that coalesces protocol conformance sections.
   Arguments.push_back("-Xlinker");
   Arguments.push_back("-T");
-  Arguments.push_back(
-      context.Args.MakeArgString(Twine(RuntimeLibPath) + "/x86_64/swift.ld"));
+#if defined(__arm__)
```

Comment:
Swift is a cross-compiler, we don't do `#if`s in the C++ sources.  Please query the target and use the correct path.  `getTriple().getArchName()`, I think, but it should also include the `gnueabihf` suffix, so please leave a fixme for that.